### PR TITLE
Generic function types

### PR DESCRIPTION
**Pull request by devoncarew**
PR created for review purposes - not intended to land.

-- edited by @jcollins-g:  Intended to land now

### CODE REVIEW

File: test/model_test.dart [hunk 1]
```diff
@@ -1493,29 +1493,41 @@ String topLevelFunction(int param1, bool param2, Cool coolBeans,
 
   group('Typedef', () {
     Typedef t;
```
Comment (devoncarew) [inline test/model_test.dart]: A minor nit - this should have been named better than `t` originally, but now that we have a 2nd related variable, perhaps a longer name  (`typedef`?)

**Comment by devoncarew** [comment]
@jcollins-g, lgtm

(hard to review w/ so many changes in generated files!)

**Comment by devoncarew** [comment]
(test failures are due to a local pub path override - they are expected)

**Comment by jcollins-g** [comment]
Let's try this again!  Generic function types can now work since analyzer 0.30 is integrated at head.

**Pull request merged**